s3/src/main/java/io/ddf/s3/S3DDFManager.java

```diff
@@ -118,15 +127,34 @@ public DataFormat getDataFormat(S3DDF s3DDF) throws DDFException {
    * @return The list of file names (TODO: should we return more info here.)
    * @brief List all the files (including directories under one path)
    */
-  public List<String> listFiles(String bucket, String key) {
+  public List<String> listFiles(String bucket, String key) throws DDFException {
```

Comment:
`key` here is not really `key` but more like `path` or `keyPrefix` right? Can it be optional?